### PR TITLE
Few code refactoring for better use of efficient NumPy functions

### PR DESCRIPTION
**Pull request by maldil**
## Why are these changes needed?
Hi Maintainers of this repo,
Thank you very much for your excellent work,
I am new to this repository.
I am a researcher studying the best practices of evolving data science codes. According to our findings after examining 1000 data science repositories, migration calculations to highly optimized APIs from `numpy` is a widespread evolution practice among developers since it improves performance as well as code quality. I found that this PR's code may leverage NumPy methods more effectively and be more succinct.

This PR is a minor contribution compared to all the hard work that you have done in this repo. However, I am hoping that it will enhance code quality and, hopefully, performance.

## Related issue number
This is a refactoring.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by gjoliver — APPROVED** [review]
for RLlib test.

**Review by krfricke — APPROVED** [review]
Thanks! I've merged master to resolve CI failures and will merge once tests pass.

**Comment by smorad** [comment]
It's not clear to me that this commit provides any benefit. First of all, this will hide divide-by-zero errors (`np.mean([])` returns `NaN` where `sum([]) / len([])` raises a `ZeroDivisionError`). Secondly, I assume that `np.mean` will convert an iterable into a `np.ndarray`, then run the mean operation, which I doubt is more efficient unless the list is at least a few hundred elements long.

File: rllib/examples/env/utils/mixins.py [hunk 1]
```diff
@@ -27,10 +28,10 @@ def _reset_info(self):
...
     def _get_episode_info(self):
         return {
-            "CC": sum(self.cc_count) / len(self.cc_count),
-            "DD": sum(self.dd_count) / len(self.dd_count),
-            "CD": sum(self.cd_count) / len(self.cd_count),
-            "DC": sum(self.dc_count) / len(self.dc_count),
+            "CC": np.mean(self.cc_count),
+            "DD": np.mean(self.dd_count),
+            "CD": np.mean(self.cd_count),
+            "DC": np.mean(self.dc_count),
```
Comment (krfricke) [inline rllib/examples/env/utils/mixins.py:34]: At least here we should probably continue to work with primitives

```suggestion
            "CC": np.mean(self.cc_count).item(),
            "DD": np.mean(self.dd_count).item(),
            "CD": np.mean(self.cd_count).item(),
            "DC": np.mean(self.dc_count).item(),
```

**Review by krfricke — COMMENTED** [review]
The test failures are known, if you merge latest master once more they should be resolved.

Let's work with primitives in rllib, I think for the other tests we should be fine. But let me know if you have strong concerns @smorad

**Comment by sven1977** [comment]
Hey @maldil , thanks for this great PR!

**Pull request merged**